src/Auth0.php

```diff
@@ -664,13 +661,13 @@ public function decodeIdToken(string $idToken, array $verifierOptions = []) : ar
         }
 
         $verifierOptions = $verifierOptions + [
-            // Set a custom leeway if one was passed to the constructor.
             'leeway' => $this->idTokenLeeway,
+        // Set a custom leeway if one was passed to the constructor.
             'max_age' => $this->transientHandler->getOnce('max_age') ?? $this->maxAge,
+            self::TRANSIENT_NONCE_KEY => $this->transientHandler->getOnce(self::TRANSIENT_NONCE_KEY)
         ];
 
-        $verifierOptions[self::TRANSIENT_NONCE_KEY] = $this->transientHandler->getOnce(self::TRANSIENT_NONCE_KEY);
-        if (empty( $verifierOptions[self::TRANSIENT_NONCE_KEY] )) {
+        if (false !== $verifierOptions[self::TRANSIENT_NONCE_KEY] && empty( $verifierOptions[self::TRANSIENT_NONCE_KEY] )) {
```

Comment:
Setting the nonce key to `false` will now skip nonce checking in the ID token before setting.